internal/querycoordv2/utils/util.go

```diff
@@ -46,23 +46,25 @@ func CheckNodeAvailable(nodeID int64, info *session.NodeInfo) error {
 // 3. The last heartbeat response time is within HeartbeatAvailableInterval for all QueryNodes(include leader) in the distribution
 // 4. All segments of the shard in target should be in the distribution
 func CheckLeaderAvailable(nodeMgr *session.NodeManager, targetMgr meta.TargetManagerInterface, leader *meta.LeaderView) error {
-	log := log.Ctx(context.TODO()).
-		WithRateGroup("utils.CheckLeaderAvailable", 1, 60).
-		With(zap.Int64("leaderID", leader.ID))
-	info := nodeMgr.Get(leader.ID)
+	log := func() *log.MLogger {
+		return log.Ctx(context.TODO()).
+			WithRateGroup("utils.CheckLeaderAvailable", 1, 60).
+			With(zap.Int64("leaderID", leader.ID))
+	}
```

Comment:
`log.With` cost more than expected, call `log.With` in each log isn't a good choice.

Reply:
It is only triggered when an error occurs; Log.With() will be called less frequently compared to the previous implementation in the CheckLeaderAvailable method.